libsolidity/formal/BMC.cpp

```diff
@@ -1192,6 +1194,15 @@ BMC::checkSatisfiableAndGenerateModel(vector<smtutil::Expression> const& _expres
 	vector<string> values;
 	try
 	{
+		if (m_settings.printQuery)
+		{
```

Comment:
We should check here if `m_settings.solvers.smtlib2` is enabled, and if not issue a warning and not print the dump.

Reply:
@leonardoalt I did that but I had some mixed feelings about it. I know it's somehow convoluted but this check was done in `SMTPortfolio` constructor.